test/parallel/test-fs-cp.mjs

```diff
@@ -38,6 +38,28 @@ function nextdir() {
   assertDirEquivalent(src, dest);
 }
 
+// It copies a nested folder structure with mode flags.
+// This test is based on fs.promises.copyFile() with `COPYFILE_FICLONE_FORCE`.
+{
+  const src = './test/fixtures/copy/kitchen-sink';
+  const dest = nextdir();
+  try {
+    cpSync(src, dest, mustNotMutateObjectDeep({
+      recursive: true,
+      mode: fs.constants.COPYFILE_FICLONE_FORCE,
+    }));
+    assertDirEquivalent(src, dest);
+    // If the platform support `COPYFILE_FICLONE_FORCE` operation,
+    // it should reach to here.
+  } catch (err) {
```

Comment:
This catch block also catches assertDirEquivalent failures, doesn't it? Better to move that outside the block.

(Also applies to code further below.)

Reply:
> This catch block also catches assertDirEquivalent failures, doesn't it?

Yes, it is. I addressed them in https://github.com/nodejs/node/pull/47084/commits/3451bbed1135748aaa54643ab0c96679fc7518db.